dianna/methods/shap.py

```diff
@@ -1,21 +1,223 @@
 import numpy as np
+import shap
+import onnx
+from skimage.segmentation import slic
+from onnx_tf.backend import prepare  # onnx to tf model converter&runner
+import warnings
 
 
-class SHAP:
+class KernelSHAP:
     """
-    SHAP implementation based on captum https://github.com/pytorch/captum
+    Kernel SHAP implementation based on shap https://github.com/slundberg/shap
     """
-    def __init__(self, shap_arg_1=0, shap_arg_2=0):
+
+    def __init__(self):
+        """KernelSHAP initializer.
+
         """
-        Example constructor for testing.
+
+    def _segment_image(
+        self,
+        image,
+        n_segments,
+        compactness,
+        max_num_iter,
+        sigma,
+        spacing,
+        convert2lab,
+        enforce_connectivity,
+        min_size_factor,
+        max_size_factor,
+        slic_zero,
+        start_label,
+        mask,
+        channel_axis_first,
```

Comment:
To simplify this function, just replace all these explicit parameters with `channels_axis_first, **kwargs` and pass everything on to `slic` below as `skimage.segmentation.slic(image=image, **kwargs)` after popping `image` from `kwargs` to handle it separately.